contest/contest_info.py

```diff
@@ -175,6 +178,10 @@ def can_delete_contest(user, contest):
 HAS_ATTENDED = "has_attended"
 IS_ADMIN = "is_admin"
 OK = "OK"
+
+'''
+return error or success message
+'''
 def can_register_return_status(user, contest):
     if not user.is_authenticated():
```

Comment:
the function is to judge can regitster or not. Anonymous user still need to use this function at template level.